include/distributed_graph_generator.h

```diff
@@ -153,9 +171,6 @@ class distributed_graph_generator {
 	// they are executed in the same order on every node.
 	std::unordered_map<collective_group_id, command_id> m_last_collective_commands;
 
-	// Side effects on the same host object create true dependencies between task commands, so we track the last effect per host object.
-	side_effect_map m_host_object_last_effects;
```

Comment:
The `side_effect_map` alias on line 128 can be dropped then (which is good, because we already have a different type with the same name in task.h!).

Reply:
Done.